contracts/aave/AaveV2Contract.sol

```diff
@@ -0,0 +1,173 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import "hardhat/console.sol";
+import "../libraries/Errors.sol";
+import "../interfaces/IPoolV2.sol";
+import "../interfaces/IWETHGateway.sol";
+import "../libraries/PercentageMath.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";
+
+contract AaveWrapper is Ownable {
+    using PercentageMath for uint256;
+
+    struct Deposits {
+        uint256 amount;
+        uint256 price;
+    }
+
+    struct Assets {
+        address asset;
+        address priceOracle;
+    }
+
+    //----------------------------//
+    //        State Variable      //
+    //----------------------------//
+    mapping(address => address) public depositAssets;
+
+    uint256 public borrowRatio = 200; // 20% of 1000
+    uint256 private constant USD_DECIMALS = 8;
+    uint256 private constant ETH_DECIMALS = 18;
+
+    IPoolV2 public constant POOL =
+        IPoolV2(0x7d2768dE32b0b80b7a3454c06BdAc94A69DDc7A9);
+
+    IWETHGateway public constant WETHGATEWAY =
+        IWETHGateway(0xcc9a0B7c43DC2a5F023Bb9b738E45B0Ef6B06E04);
+
+    IERC20 public constant BORROW_ASSET =
+        IERC20(0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48); // USDC
+
+    //----------------------------//
+    //          Events            //
+    //----------------------------//
+
+    event AddAsset(address indexed _asset, address _oracle);
+    event BorrowRatioChanged(uint256 indexed ratio);
+
+    //----------------------------//
+    //          Modifiers         //
+    //----------------------------//
+
+    modifier isValid(address _asset) {
+        require(depositAssets[_asset] == address(0));
+        _;
+    }
+
+    //----------------------------//
+    //       View Functions       //
+    //----------------------------//
+
+    /**
+     * @dev fetch the latest price of the asset from oracle
+     * @param _asset address for which we need price
+     */
+    function getLatestPrice(address _asset) public view returns (int256 price) {
+        AggregatorV3Interface priceFeed = AggregatorV3Interface(_asset);
+        (, price, , , ) = priceFeed.latestRoundData();
+    }
+
+    function nomalize(uint256 amount) public view returns (uint256) {
+        return amount * 10**(ETH_DECIMALS - USD_DECIMALS);
+    }
+
+    //----------------------------//
+    //     Mutation Functions     //
+    //----------------------------//
+
+    /**
+     * @dev Only admin can add new supply assets and their oracles
+     * @param _asset will be address of the asset
+     * @param _priceOracle will the chainlink asset price oracle
+     * for fetching asset current price
+     */
+    function addDepositAsset(
+        address _asset,
+        address _priceOracle
+    ) public onlyOwner isValid( _asset) {
+        depositAssets[_asset] = _priceOracle;
+        emit AddAsset(_asset,_priceOracle);
```

Comment:
Also one request: 
For which ever "major" or "import" function that's emitting something (like this addDeposit Asset), can we write an example of what will be emitted upon successful execution in the code itself as comments ?